pyrogram/client/methods/chats/set_chat_slowmode.py

```diff
@@ -0,0 +1,69 @@
+# Pyrogram - Telegram MTProto API Client Library for Python
+# Copyright (C) 2017-2019 Dan Tès <https://github.com/delivrance>
+#
+# This file is part of Pyrogram.
+#
+# Pyrogram is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Pyrogram is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Pyrogram.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Union
+
+from pyrogram.api import functions, types
+from ...ext import BaseClient
+
+
+class SetChatSlowmode(BaseClient):
+    def set_chat_slowmode(
+        self,
+        chat_id: Union[int, str],
+        seconds: int
+    ) -> bool:
+        """Set the slow mode in a supergroup. If it is enabled, users can only send one message in the specified time.
+        Valid numbers of seconds are: 0 (disabled), 10, 30, 60, 300, 900, 3600
```

Comment:
As the values for this parameter are restricted to a number of choices, maybe it would make sense to abstract them into an enumeration so that users of autocompletion can immediately see the allowed values without concerning the documentation. Of course using the enum should be optional, meaning that if we use an `IntEnum` it should also be possible to pass an `int`. However, naming in Python is quite restrictive and class member symbols cannot start with numeric characters...

The only things I could come up with were

```
class SlowModeDuration(IntEnum):
    _10s = 10
    _30s = 30
    _60s = 60
    _300s = 300
    _900s = 900
    _3600s = 3600
```

and

```
class SlowModeDurationSeconds(IntEnum):
    ten_seconds = 10
    thirty_seconds = 30
    one_minute = 60
    five_minutes = 300
    fifteen_minutes = 900
    one_hour = 3600
```

neither of which I'm particularly fond of. 

Maybe someone else has an idea...

Reply:
Enums are definitely a good idea. The main reason they are not used already is for consistency with other parts in the library. We need quite a lot of enums, each dedicated to a specific part (chat actions, chat member filters, slowmo durations, etc...). This should be tracked in a whole new issue.